test/worker.test.js

```
@@ -0,0 +1,151 @@
+      const data = await res.json();
+      expect(data.ip).toBe('127.0.0.1');
+      expect(data.city).toBe('Test City');
+    });
+  });
+
+  describe('Rate Limiting', () => {
+    it('should allow requests under the rate limit', async () => {
+      mockKV.get.mockResolvedValueOnce('10'); // Current count
+
+      const { req, ctx } = createRequest('GET', '/ping');
+      const res = await worker.fetch(req, env, ctx);
```

ESLint

[error] 'env' is not defined.

(no-undef)